CCHMapClusterController/CCHMapClusterOperation.m

```diff
@@ -97,20 +101,237 @@ + (MKMapRect)gridMapRectForMapRect:(MKMapRect)mapRect withCellMapSize:(double)ce
     return gridMapRect;
 }
 
+
+- (MKZoomScale) currentZoomScale
+{
+	CGSize screenSize = self.mapView.bounds.size;
+	MKMapRect mapRect = self.mapView.visibleMapRect;
+	MKZoomScale zoomScale = screenSize.width / mapRect.size.width;
+	return zoomScale;
+}
+
+
++ (double) distanceSquared:(MKMapPoint)a b:(MKMapPoint)b {
+	return (a.x - b.x) * (a.x - b.x) + (a.y - b.y) * (a.y - b.y);
+}
+
+
++ (MKMapRect) createBoundsFromSpan:(MKMapPoint)p span:(double)span {
+	double halfSpan = span / 2;
+	return MKMapRectMake(p.x - halfSpan, p.y - halfSpan, span, span);
+}
+
+
+
 - (void)start
 {
+	switch (self.clusterMethod) {
+		case ClusterMethodGridBased:
+			[self start_grid];
+			break;
+		case ClusterMethodDistanceBased:
+			[self start_distance];
+			break;
+	};
+}
+
+
+
+// A simple clustering algorithm with O(nlog n) performance. Resulting clusters are not
+// hierarchical. Contributed by GitHub user __amishjake__.
+// Steps:
+// 1. Iterate over candidate annotations.
+// 2. Create/re-use a CCHMapClusterAnnotation with the center of the annotation.
+// 3. Add all annotations that are within a certain distance to the cluster.
+// 4. Move any annotations out of an existing cluster if they are closer to another cluster.
+// 5. Remove those items from the list of candidate clusters.
+// 6. Add/remove items on the actual map as in the grid algorithm.
+//
+- (void) start_distance
```

Comment:
I guess in objective c we follow the naming convention to use camel case or method names? `startDistance`